lib/games_services.dart

```diff
@@ -1,56 +1,100 @@
+  String toString() => 'GameServicesScoreException: ${this.baseException}';
+};
+class GameServicesLeaderboardException extends GamesServicesException {
+  String toString() => 'GameServicesLeaderboardException: ${this.baseException}';
+};
+class GameServicesLoginException extends GamesServicesException {
+  String toString() => 'GameServicesLoginException: ${this.baseException}';
+};
+class GameServicesUnknownException extends GamesServicesException {
+  String toString() => 'GameServicesUnknownException: ${this.baseException}';
+};
+
 class GamesServices {
   static const MethodChannel _channel = const MethodChannel('games_services');
 
+  static Future<void> _wrapPlatformException(AsyncError asyncError) async {
+    switch (asyncError.error.code) {
+      case 'SUBMIT_ACHIEVEMENT_ERROR':
+        throw GameServicesAchievementException(asyncError.error);
+      case 'SUBMIT_SCORE_ERROR':
+        throw GameServicesScoreException(asyncError.error);
+      case 'VIEW_ACHIEVEMENT_ERROR':
+        throw GameServicesAchievementException(asyncError.error);
+      case 'VIEW_LEADERBOARD_ERROR':
+        throw GameServicesLeaderboardException(asyncError.error);
+      case 'LOGIN_ERROR':
+        throw GameServicesLoginException(asyncError.error);
+      default:
+        throw GameServicesUnknownException(asyncError.error);
+      }
+  }
+
   /// Unlock an [achievement].
   /// [Achievement] takes three parameters:
   /// [androidID] the achievement id for android.
   /// [iOSID] the achievement id for iOS.
   /// [percentComplete] the completion percent of the achievement, this parameter is
   /// optional in case of iOS.
-  static Future<String> unlock({achievement: Achievement}) async {
-    return await _channel.invokeMethod("unlock", {
-      "achievementID": achievement.id,
-      "percentComplete": achievement.percentComplete,
-    });
+  static Future<void> unlock({achievement: Achievement}) async {
+    return _channel.invokeMethod('unlock', {
+      'achievementID': achievement.id,
+      'percentComplete': achievement.percentComplete,
+    }).catchError(_wrapPlatformException, test (e): => e is PlatformException);
   }
 
   /// Submit a [score] to specific leader board.
   /// [Score] takes three parameters:
   /// [androidLeaderboardID] the leader board id that you want to send the score for in case of android.
   /// [iOSLeaderboardID] the leader board id that you want to send the score for in case of iOS.
   /// [value] the score.
-  static Future<String> submitScore({score: Score}) async {
-    return await _channel.invokeMethod("submitScore", {
-      "leaderboardID": score.leaderboardID,
-      "value": score.value,
-    });
+  static Future<void> submitScore({score: Score}) async {
+    return _channel.invokeMethod('submitScore', {
+      'leaderboardID': score.leaderboardID,
+      'value': score.value,
+    }).catchError(_wrapPlatformException, test (e): => e is PlatformException);
```

Comment:
Lol, that's an obvious typo

Reply:
Why are awaits removed from all the invokeMethod calls?

Reply:
This is over a year old, I don't remember what was going on. Maybe why await when you can defer through promise?
Not sure, I don't remember if I ever did compile this.